src/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class ExceptionTests
+    {
+        [Fact]
+        public static void RootException()
+        {
+            bool exceptionThrown = false;
+            try
+            {
+                int i2 = JsonSerializer.Parse<int>("12bad");
+            }
+            catch (JsonSerializationException e)
+            {
+                exceptionThrown = true;
+
+                Assert.Equal(0, e.LineNumber);
+                Assert.Equal(2, e.BytePositionInLine);
+                Assert.Equal("[System.Int32]", e.Path);
+            }
+
+            Assert.True(exceptionThrown);
+        }
+
+        [Fact]
+        public static void RethrownJsonReaderException()
+        {
+            bool exceptionThrown = false;
+            string json = Encoding.UTF8.GetString(BasicCompany.s_data);
+
+            json = json.Replace(@"""zip"" : 98052", @"""zip"" : bad");
```

Comment:
Presumably there are other cases where the `JsonReaderException` is being wrapped by a `JsonSerializationException`, or is this it? Should we add tests for some more of those cases?

Reply:
Yes there are but they are handled by other tests. This test just picks one case and ensures that the code that wraps the reader exception has the proper text. The other cases use the same code to wrap so they don't all need to be testsed.